XMonad/Actions/RepeatAction.hs

```diff
@@ -0,0 +1,60 @@
+-----------------------------------------------------------------------------
+-- |
+-- Module      :  XMonad.Actions.RepeatAction
+-- Description :  Repeat the last performed action.
+-- Copyright   :  (c) 2022 Martin Kozlovsky
+-- License     :  BSD3-style (see LICENSE)
+--
+-- Maintainer  :  <kozlovsky.m7@gmail.com>
+-- Stability   :  unstable
+-- Portability :  not portable
+--
+-- Ability to repeat the last action.
+--
+-----------------------------------------------------------------------------
+
+module XMonad.Actions.RepeatAction (
+        -- * Usage
+        -- $usage
+          XMonad.Actions.RepeatAction.rememberAction
+        , XMonad.Actions.RepeatAction.rememberActions
+        , XMonad.Actions.RepeatAction.repeatLast
+) where
+
+import XMonad
+
+import qualified XMonad.Util.ExtensibleState as XS
+
+-- $usage
+--
+-- You can use this module with the following in your @xmonad.hs@:
+--
+-- > import XMonad.Actions.RepeatAction
+--
+-- Then use the 'rememberActions' on the list of your keybindings and
+-- add a dedicated key to run the last performed action:
+--
+-- > myKeys = ((modm, xK_period), repeatLast) : rememberActions [((modm, xK_c), kill), ...]
+--
+-- / Be careful not to include the key binding for the repetition in the remembered actions! /
+--
+-- For detailed instructions on editing your key bindings, see
+-- "XMonad.Doc.Extending#Editing_key_bindings".
+
+newtype LastAction = LastAction { runLastAction :: X () }
+
+instance ExtensionClass LastAction where
```

Comment:
Indeed, that would be quite bad.  Perhaps then a better interface would be to not expose `repeatLast` individually, but only in combination with `rememberActions`?  Something like `rememberActions :: (KeyMask, KeySym) -> [((KeyMask, KeySym), X ())] -> XConfig a -> XConfig a`?  This could then be used as a combinator.  Might also be worth having a `rememberActionsP :: String -> [(String, X ())] -> XConfig l -> XConfig l`